lib/rules/alpha-value-notation/__tests__/index.mjs

```diff
@@ -541,3 +581,218 @@ testRuleConfigs({
+				{
+					message: messages.expected('70%', '0.7'),
+					line: 1,
+					column: 64,
+					endLine: 1,
+					endColumn: 67,
+					fix: { range: [63, 66], text: '0.7' },
+				},
+			],
+		},
+		{
+			code: 'a { opacity: 10%; shape-image-threshold: 50%; }',
+			warnings: [
+				{
+					message: messages.expected('10%', '0.1'),
+					line: 1,
+					column: 14,
+					endLine: 1,
+					endColumn: 17,
+					fix: { range: [13, 16], text: '0.1' },
+				},
+				{
+					message: messages.expected('50%', '0.5'),
+					line: 1,
+					column: 42,
+					endLine: 1,
+					endColumn: 45,
+					fix: { range: [41, 44], text: '0.5' },
+				},
+			],
+		},
+		{
+			code: '@keyframes name { color: rgba(0, 0, 0, 45%); }',
+			message: messages.expected('45%', '0.45'),
+			line: 1,
+			column: 40,
+			endLine: 1,
+			endColumn: 43,
+			fix: { range: [39, 42], text: '0.45' },
+		},
+		{
+			code: `a { color: red; opacity: 30%; }`,
+			message: messages.expected('30%', '0.3'),
+			line: 1,
+			column: 26,
+			endLine: 1,
+			endColumn: 29,
+			fix: { range: [25, 28], text: '0.3' },
+		},
+	],
+});
+
+// Percentage with computeEditInfo
+testRule({
+	ruleName,
+	config: ['percentage'],
+	computeEditInfo: true,
+
+	accept: [
+		{
+			code: `@import "foo.css";`,
+		},
+	],
+
+	reject: [
+		{
+			code: 'a { opacity: 0.1; }',
+			message: messages.expected('0.1', '10%'),
+			line: 1,
+			column: 14,
+			endLine: 1,
+			endColumn: 17,
+			fix: { range: [13, 16], text: '10%' },
+		},
+		{
+			code: 'a { shape-image-threshold: 0.5 }',
+			message: messages.expected('0.5', '50%'),
+			line: 1,
+			column: 28,
+			endLine: 1,
+			endColumn: 31,
+			fix: { range: [27, 30], text: '50%' },
+		},
+		{
+			code: 'feDropShadow { flood-opacity: 0.5 }',
+			message: messages.expected('0.5', '50%'),
+			line: 1,
+			column: 31,
+			endLine: 1,
+			endColumn: 34,
+			fix: { range: [30, 33], text: '50%' },
+		},
+		{
+			code: 'a { color: rgba(0, 0, 0, 0.5) }',
+			message: messages.expected('0.5', '50%'),
+			line: 1,
+			column: 26,
+			endLine: 1,
+			endColumn: 29,
+			fix: { range: [25, 28], text: '50%' },
+		},
+		{
+			code: 'a { color: rgba(0, 0, 0, 0.5); background-color: rgba(0, 0, 0, 0.7); }',
+			warnings: [
+				{
+					message: messages.expected('0.5', '50%'),
+					line: 1,
+					column: 26,
+					endLine: 1,
+					endColumn: 29,
+					fix: { range: [25, 28], text: '50%' },
+				},
+				{
+					message: messages.expected('0.7', '70%'),
+					line: 1,
+					column: 64,
+					endLine: 1,
+					endColumn: 67,
+					fix: { range: [63, 66], text: '70%' },
+				},
+			],
+		},
+		{
+			code: 'a { opacity: 0.1; shape-image-threshold: 0.5; }',
+			warnings: [
+				{
+					message: messages.expected('0.1', '10%'),
+					line: 1,
+					column: 14,
+					endLine: 1,
+					endColumn: 17,
+					fix: { range: [13, 16], text: '10%' },
+				},
+				{
+					message: messages.expected('0.5', '50%'),
+					line: 1,
+					column: 42,
+					endLine: 1,
+					endColumn: 45,
+					fix: undefined,
```

Comment:
```suggestion
					fix: { range: [41, 44], text: '50%' },
```

see https://github.com/stylelint/stylelint/pull/8369/files#r1970638806 for details